Use message queue gem

bishboria opened the pull request:
This PR is part of a push to make use the [message consumer gem](https://github.com/alphagov/govuk_message_queue_consumer) in all apps that need to listen to the publishing queue. 
- See https://github.com/alphagov/email-alert-service/pull/42 for the `email-alert-service` implementation.
- See https://github.com/alphagov/panopticon/pull/307 for the `panopticon` implementation.

Todo before merge:
- [x] ENV variables should be available for content-register
- [x] Use rubygems `govuk_message_queue_consumer` 2.X instead of github version

Trello: https://trello.com/c/z8b6VmCU

alext commented on Gemfile:
```diff
@@ -4,7 +4,7 @@ gem 'rails', '4.1.11'
 
 gem 'pg', '0.17.1'
-gem 'bunny', '~> 1.5'
+gem 'message-queue-consumer', '0.9.1'
```
I wonder about the naming of this gem. Given this is published in a public namespace, it seems a little antisocial to use such a general name for this given it's somewhat specific to our use of RabbitMQ.  It would probably be better to prefix the name with 'govuk' or something like that.

rboulton replied:
Although I named the repository, I think I agree.  It will be easier to rename now than later, and I think we should rename the Gem.

tijmenb commented:
@rboulton @alext 

I've squashed and updated this PR. It should be good to merge, but I hold off doing that until we've added the panopticon PR. That might surface some issues relevant to this repo. But if you can review this PR in the meantime I'd be happy.

rboulton commented on lib/tasks/message_queue.rake:
```diff
@@ -1,16 +1,9 @@
-    end
+  desc "Run worker to consume messages from RabbitMQ"
+  task :consumer => [:environment] do
+    puts "Starting message queue..."
```
I think the old message ("Starting message consumer") was more accurate - the message queue is a thing, not a process.  Actually, most precise would be to say "Starting message queue consumer".

tijmenb replied:
True!

While were here - I didn't see the use in `rescue SignalException => e` - any thoughts?

alext replied:
It's there to prevent spurious errbit reports when the service is restarted. The Airbrake gem inserts a handler to report any exceptions raised form rake tasks, and Bunny raises this exception when the process it terminated with a signal by upstart.

rboulton replied:
It seems useful to ensure that there's output in the logs when the consumer exits on a signal (usually sigterm, probably).  However, it's a bit ugly if all the consumers have to wrap `.run` in that rescue.  Can we put that rescue into the .run method on the Gem?

tijmenb replied:
That would make it something for the gem to handle. Agree?

alext replied:
If the gem is providing the rake task, there's no way for the app to be able to handle it really.

rboulton commented:
One tiny (non-merge-blocking) nit, but otherwise this is fine.

tijmenb commented:
Updated PR to use the new environment var version of the gem.

tijmenb committed:
Use govuk_message_queue_gem

This commit replaces this app's custom message queue with the new
standard `govuk_message_queue_gem`. There should be no changes in
behaviour.

tijmenb commented:
I've squashed & rebased the commits here. I think this should work, but haven't done a manual test.

I'm closing this and making a new ticket on Trello: 

https://trello.com/c/rCUEgyjz/413-make-content-register-use-the-message-queue-gem

We'd like to focus on getting panopticon running with this gem first.
